src/dfcx_scrapi/tools/test_cases_util.py

```diff
@@ -0,0 +1,704 @@
+            is_webhook_enabled: if enable then True else False
+            agent_id: The formatted CX Agent ID
+        Returns:
+            list of the updated test caess
+        """
+        test_cases_list = self._global_test_cases.list_test_cases(
+            agent_id=self.agent_id, include_conversation_turns=True)
+        updated_test_cases_list = []
+        for test_case in test_cases_list:
+            if test_case.name in test_case_ids:
+                test_case.test_case_conversation_turns = (
+                    self._toggle_webhooks_in_convturns(
+                        conv_turns=test_case.test_case_conversation_turns,
+                        is_webhook_enabled=is_webhook_enabled)
+                    )
+                updated_test_cases_list.append(test_case)
+
+        return updated_test_cases_list
+
+    @staticmethod
+    def _reverse_dict(dict_):
+        """Utility method that reverse the dict keys and values
+            Args:
+                dict_: a dictionary
+            Returns:
+                a reversed dictionary
+        """
+
+        return {v: k for k, v in dict_.items()}
+
+    def _get_commons_config(
+        self,
+        agent_id: str) -> Dict[str, Dict[str, str]]:
+        """This method map flows, pages, and intents
+            are in the dictionary format
+          Arg:
+            agent_id: The formatted CX Agent ID
+          Returns:
+            a dictionary of flows, pages, and intents
+        """
+
+        commons_config = {}
+        commons_config["agent_id"] = agent_id
+        dfcx_flows = flows.Flows(creds=self.creds, agent_id=agent_id)
+        dfcx_pages = pages.Pages(creds=self.creds)
+        dfcx_intents = intents.Intents(creds=self.creds, agent_id=agent_id)
+        dfcx_testcases = test_cases.TestCases(
+            creds=self.creds, agent_id=agent_id)
+        dfcx_intents_map = dfcx_intents.get_intents_map(agent_id=agent_id)
+        dfcx_flows_map = dfcx_flows.get_flows_map(agent_id=agent_id)
+        dfcx_pages_map = {}
+        for flow_id in dfcx_flows_map.keys():
+            dfcx_pages_map[flow_id] = dfcx_pages.get_pages_map(flow_id=flow_id)
+        commons_config["test_cases"] = dfcx_testcases
+        commons_config["flows"] = dfcx_flows
+        commons_config["pages"] = dfcx_pages
+        commons_config["intents"] = dfcx_intents
+        commons_config["flows_map"] = dfcx_flows_map
+        commons_config["pages_map"] = dfcx_pages_map
+        commons_config["intents_map"] = dfcx_intents_map
+        commons_config["flows_map_reverse"] = (
+            self._reverse_dict(dict_=commons_config["flows_map"]))
+        commons_config["intents_map_reverse"] = (
+            self._reverse_dict(dict_=commons_config["intents_map"]))
+        commons_config["pages_map_reverse"] = (
+            {f_id:self._reverse_dict(dict_=pages_map) \
+             for f_id, pages_map in dfcx_pages_map.items()})
+
+        return commons_config
+
+    def _get_new_test_config(
```

Comment:
@jkshj21 for this method, if both of your if statements fail (i.e. neither flow or page is found) then the method will return an empty TestConfig. Is that the intention here?

Or should it always find some value and return that config?

If it's the former, can you please update the docstring to explain this?
i.e. "If a TestConfig is found, it will return that, otherwise an empty TestConfig is returned."

Reply:
Q: for this method, if both of your if statements fail (i.e. neither flow or page is found) then the method will return an empty TestConfig. Is that the intention here? Or should it always find some value and return that config?

Josh: Yes, that is the right intention. The reason for returning a "None" is to satisfy the below condition.
at line 516, "if not all([test_config, conv_turns, last_test_conv_turns])"

When we are converting a test_case from a source to a target agent, there are many cases when it fails. Testconfig is one of the cases that might fail. I can output an error and stop this function to run, (returns test_config or raise an error) but I thought it would be ideal if we just output an error and let the function run, because most likely, the users will convert more than one test case. 

While the test case conversion is looping, the function will stop the process if it fails then it might convert some of the test cases without looping all the way.
for test_case in list_test_cases:
    test_cases_util.convert_test_case(test_case)

I added the docstring to explain this "If a TestConfig is found, it will return that, otherwise an empty TestConfig is returned."

Reply:
Thanks for the detailed explanation! Updating the docstring should be sufficient.